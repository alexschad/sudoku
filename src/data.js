export const rows = [
  [0, 1, 2, 3, 4, 5, 6, 7, 8],
  [9, 10, 11, 12, 13, 14, 15, 16, 17],
  [18, 19, 20, 21, 22, 23, 24, 25, 26],
  [27, 28, 29, 30, 31, 32, 33, 34, 35],
  [36, 37, 38, 39, 40, 41, 42, 43, 44],
  [45, 46, 47, 48, 49, 50, 51, 52, 53],
  [54, 55, 56, 57, 58, 59, 60, 61, 62],
  [63, 64, 65, 66, 67, 68, 69, 70, 71],
  [72, 73, 74, 75, 76, 77, 78, 79, 80],
];
export const cols = [
  [0, 9, 18, 27, 36, 45, 54, 63, 72],
  [1, 10, 19, 28, 37, 46, 55, 64, 73],
  [2, 11, 20, 29, 38, 47, 56, 65, 74],
  [3, 12, 21, 30, 39, 48, 57, 66, 75],
  [4, 13, 22, 31, 40, 49, 58, 67, 76],
  [5, 14, 23, 32, 41, 50, 59, 68, 77],
  [6, 15, 24, 33, 42, 51, 60, 69, 78],
  [7, 16, 25, 34, 43, 52, 61, 70, 79],
  [8, 17, 26, 35, 44, 53, 62, 71, 80],
];
export const squares = [
  [0, 1, 2, 9, 10, 11, 18, 19, 20],
  [3, 4, 5, 12, 13, 14, 21, 22, 23],
  [6, 7, 8, 15, 16, 17, 24, 25, 26],
  [27, 28, 29, 36, 37, 38, 45, 46, 47],
  [30, 31, 32, 39, 40, 41, 48, 49, 50],
  [33, 34, 35, 42, 43, 44, 51, 52, 53],
  [54, 55, 56, 63, 64, 65, 72, 73, 74],
  [57, 58, 59, 66, 67, 68, 75, 76, 77],
  [60, 61, 62, 69, 70, 71, 78, 79, 80],
];

// prettier-ignore
export const sudokus = [
  [null,null,2,null,null,4,8,9,null,null,5,null,8,null,null,3,null,7,7,null,null,null,5,null,null,4,6,null,6,null,null,null,null,null,null,4,null,null,1,null,9,null,5,null,null,9,null,null,null,null,null,null,8,null,6,3,null,null,1,null,null,null,8,1,null,4,null,null,8,null,2,null,null,9,8,4,null,null,1,null,null],
  [8,null,null,9,null,7,null,null,3,null,null,null,1,4,6,null,null,null,null,null,9,null,null,null,1,null,null,2,3,null,6,null,5,null,1,7,null,6,null,null,null,null,null,4,null,9,1,null,4,null,8,null,3,6,null,null,3,null,null,null,6,null,null,null,null,null,5,3,9,null,null,null,4,null,null,7,null,2,null,null,1],
  [3,7,null,null,null,null,8,1,null,6,null,null,3,null,null,null,4,5,null,null,null,1,null,null,7,null,2,null,5,3,4,null,null,null,2,null,null,null,null,null,null,null,null,null,null,null,2,null,null,null,7,4,8,null,5,null,7,null,null,3,null,null,null,9,1,null,null,null,6,null,null,3,null,3,4,null,null,null,null,6,8],
  [1,null,null,2,null,5,null,null,6,null,null,null,8,7,4,null,null,null,null,null,5,null,9,null,8,null,null,8,2,null,null,null,null,null,3,5,null,5,3,null,null,null,6,7,null,4,7,null,null,null,null,null,9,8,null,null,8,null,1,null,5,null,null,null,null,null,7,8,9,null,null,null,3,null,null,5,null,2,null,null,9],
  [null,3,6,null,null,1,null,null,null,null,null,null,3,null,2,4,null,null,null,8,null,null,7,null,5,null,null,null,null,9,null,null,4,null,null,2,1,null,null,null,null,null,null,null,6,2,null,null,7,null,null,3,null,null,null,null,1,null,8,null,null,3,null,null,null,4,5,null,3,null,null,null,null,null,null,1,null,null,7,2,null],
  [4,null,null,null,1,null,9,2,null,null,5,null,null,null,null,null,null,4,null,null,null,7,9,4,3,null,5,2,null,7,null,3,null,5,null,null,5,null,4,2,null,9,6,null,1,null,null,1,null,8,null,7,null,2,3,null,2,6,4,7,null,null,null,6,null,null,null,null,null,null,1,null,null,9,5,null,2,null,null,null,3],
  [null,null,4,1,2,5,8,null,null,null,null,null,null,4,null,null,null,null,8,null,null,7,null,9,null,null,5,4,null,5,null,null,null,1,null,3,9,6,null,null,null,null,null,4,7,2,null,1,null,null,null,5,null,8,6,null,null,9,null,4,null,null,1,null,null,null,null,7,null,null,null,null,null,null,2,3,1,6,7,null,null],
  [8,9,null,null,null,4,null,null,7,3,1,5,null,null,null,null,2,null,null,6,7,null,null,null,8,null,null,null,null,null,6,4,null,null,null,1,null,null,null,3,7,2,null,null,null,7,null,null,null,8,1,null,null,null,null,null,4,null,null,null,2,8,null,null,8,null,null,null,null,1,7,5,1,null,null,5,null,null,null,9,6],
  [null,null,8,null,null,null,6,null,null,null,7,null,null,1,null,null,5,null,5,null,null,3,null,6,null,null,4,null,null,2,6,null,3,4,null,null,null,3,null,null,null,null,null,6,null,null,null,6,5,null,7,9,null,null,9,null,null,7,null,2,null,null,8,null,6,null,null,9,null,null,3,null,null,null,7,null,null,null,2,null,null],
  [null,5,null,8,null,1,null,9,null,7,3,null,null,null,null,null,5,4,8,null,null,null,3,null,null,null,1,null,null,8,3,null,2,1,null,null,null,null,null,null,null,null,null,null,null,null,null,6,7,null,4,5,null,null,1,null,null,null,5,null,null,null,9,3,8,null,null,null,null,null,1,2,null,4,null,6,null,8,null,3,null],
  [null,null,3,null,5,null,null,1,null,null,null,null,null,null,null,9,4,null,5,null,null,null,null,2,null,null,8,null,2,null,3,4,1,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,8,9,7,null,6,null,3,null,null,5,null,null,null,null,6,null,6,1,null,null,null,null,null,null,null,7,null,null,2,null,4,null,null],
];
