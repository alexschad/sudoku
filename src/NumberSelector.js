import React from 'react'
import { rows, cols, squares } from './data';

export default function NumberSelector({clicked, setClicked, selectNumber, selPos, fields}) {
  const row = rows.find(r => r.includes(clicked));
  const col = cols.find(c => c.includes(clicked));
  const square = squares.find(s => s.includes(clicked));
  const usedIndices = (row && col && square) ? row.concat(col).concat(square) : [];
  const usedVals = usedIndices.map(ui => fields[ui]).filter(i => i !== null);

  const nCont = document.getElementById('numberSelectorContainer');
  const height = nCont ? Math.floor(nCont.offsetHeight / 2) : 0;
  const width = nCont ? Math.floor(nCont.offsetWidth / 2) : 0;

  return (
    <div
      id="numberSelectorContainer"
      style={{
        display: clicked!== null ? 'block' : 'none',
        top: selPos[1] - height + 'px',
        left: selPos[0] - width + 'px',
      }}
    >
      <div className="numberSelector" onMouseLeave={() => setClicked(null)}>
        <div className="selectorClose" onClick={() => selectNumber(clicked, null)}>X</div>
        {
          [1,2,3,4,5,6,7,8,9].map( n => {
            if (usedVals.includes(n)) {
              return <div key={n}></div>
            } else {
              return <div key={n} onClick={() => selectNumber(clicked, n)}>{n}</div>
            }
          })
        }
      </div>
    </div>
  )
}
