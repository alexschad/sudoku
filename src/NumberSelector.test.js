import React from 'react';
import renderer from 'react-test-renderer';
import NumberSelector from './NumberSelector';
import { sudokus } from './data';

test('it renders', () => {
  const component = renderer.create(
    <NumberSelector
    selectNumber={1}
    clicked={false}
    setClicked={() => {}}
    selPos={[0,0]}
    fields={sudokus[0]}
  />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});