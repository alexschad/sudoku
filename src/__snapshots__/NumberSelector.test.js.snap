// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  id="numberSelectorContainer"
  style={
    Object {
      "display": "block",
      "left": "0px",
      "top": "0px",
    }
  }
>
  <div
    className="numberSelector"
    onMouseLeave={[Function]}
  >
    <div
      className="selectorClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      onClick={[Function]}
    >
      1
    </div>
    <div
      onClick={[Function]}
    >
      2
    </div>
    <div
      onClick={[Function]}
    >
      3
    </div>
    <div
      onClick={[Function]}
    >
      4
    </div>
    <div
      onClick={[Function]}
    >
      5
    </div>
    <div
      onClick={[Function]}
    >
      6
    </div>
    <div
      onClick={[Function]}
    >
      7
    </div>
    <div
      onClick={[Function]}
    >
      8
    </div>
    <div
      onClick={[Function]}
    >
      9
    </div>
  </div>
</div>
`;
